src/tcl/message.rs

```diff
@@ -104,3 +138,99 @@ pub async fn receive<T: for<'a> Deserialize<'a>>(
     // return the message if everything went right
     Ok(received_message)
 }
+
+/* -------------------------------------------------------------------------- */
+/*                           Display Implementation                           */
+/* -------------------------------------------------------------------------- */
+fn format_duration(duration: Duration) -> String {
+    let secs = duration.as_secs();
+    let hours = secs / 3600;
+    let minutes = (secs % 3600) / 60;
+    let seconds = secs % 60;
+    format!("{:02}:{:02}:{:02}", hours, minutes, seconds)
+}
+
+impl Display for ProcessState {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{self:#10?}")
+    }
+}
+
+impl Display for ProcessStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "┌─ Process Status ───────────────────────────────────")?;
+        writeln!(f, "│ {:20} {}", "State:", self.status)?;
+        writeln!(
+            f,
+            "│ {:20} {}",
+            "PID:",
+            self.pid
+                .map_or("Not assigned".to_string(), |pid| pid.to_string())
+        )?;
+        writeln!(
+            f,
+            "│ {:20} {}",
+            "Started:",
+            self.start_time
+                .map_or("Not yet".to_string(), |time| format_duration(
+                    time.duration_since(UNIX_EPOCH).unwrap()
+                ))
+        )?;
```

Comment:
Maybe you could use format_duration function ?

Reply:
elle est utiliser